src/Illuminate/Contracts/Auth/Guard.php

```diff
@@ -43,8 +43,8 @@ public function validate(array $credentials = []);
     /**
      * Set the current user.
      *
-     * @param  \Illuminate\Contracts\Auth\Authenticatable  $user
+     * @param  \Illuminate\Contracts\Auth\Authenticatable|null  $user
      * @return void
      */
-    public function setUser(Authenticatable $user);
+    public function setUser(?Authenticatable $user);
```

Comment:
This unfortunately is a breaking change to the method signature and the contract.

Reply:
Thanks @driesvints that was quick! Isn't this something we can just push for the next major version then? 

It isn't possible as far as I can tell to undo a setUser call on any driver other than the session one (where you can use logout). 

What if I add a logout method on `GuardHelpers` and the contract which will just set `$this->user = null;` and then create an `actingAsGuest` which simply calls logout? That shouldn't be a breaking change, just an addition :smile: